include/onnxruntime/core/framework/stream_handles.h

```diff
@@ -131,7 +131,7 @@ class Notification {
   }
 
  protected:
-  virtual void Activate() = 0;
+  virtual void Activate() {};
```

Comment:
Instead of add default implementation in the header, could we add a "DummyStream" implementation in core framework and used it for the memory pattern case?
Because the header will be included in multiple EPs, better to make it pure virtual.